app/models/portfolio_item.rb

```diff
@@ -14,4 +14,12 @@ class PortfolioItem < ApplicationRecord
   belongs_to :portfolio
   validates :service_offering_ref, :name, :presence => true
   validates :favorite_before_type_cast, :format => { :with => /\A(true|false)\z/i }, :allow_blank => true
+
+  attribute :metadata, ActiveRecord::Type::Json.new
+
+  def metadata
+    user = Insights::API::Common::Request.current!
+
+    {:user_capabilities => PortfolioItemPolicy.new(user, self).user_capabilities}
```

Comment:
@eclarizio 
Can we figure out the Policy Class automatically
https://github.com/varvet/pundit/blob/bb09157424caf41e1bd22e218e0208d1c9548a4d/lib/pundit/policy_finder.rb#L7

Reply:
Hmm, we could, it's gonna look kinda funny though. I guess you're interested in separating this out into a mixin or something?
```
{:user_capabilities => Pundit::PolicyFinder.new(self).policy.new(user, self).user_capabilities}
```
Seems a little convoluted, I'd rather just do it manually if we're going that route:
```
{:user_capabilities => "#{self.class}Policy".constantize.new(user, self).user_capabilities}
```

Reply:
@eclarizio Should the User Capabilities be a separate class that should end up calling the Policy and build the capabilities from that. For now we could go with this.